020-Getting-started/080-astro.mdx

```diff
@@ -50,153 +51,713 @@ apiKey=YOUR_API_KEY_HERE
+<Expand title="xata init output">
+
+```sh
+$ xata init
+🦋 Initializing project... We will ask you some questions.
+
+You have a single workspace, using it by default: Phil-Leggetter-s-workspace-hc84d7
+✔ Select a database or create a new one › <Create a new database>
+✔ New database name … xata-astro
+✔ Select a region › eu-west-1
+✔ Generate code and types from your Xata database › TypeScript
+✔ Choose the output path for the generated code … src/xata.ts
+
+Setting up Xata...
+
+Created Xata config: .xatarc
 
-Your `.env` file should look something like this.
+Creating .env file
+set XATA_API_KEY=xau_**********************************
+set XATA_BRANCH=main
 
-```bash
-# You should now have a ~/my-xata-project/.env file
+i Running npm install --save @xata.io/client
+
+added 1 package, and audited 462 packages in 2s
+
+181 packages are looking for funding
+run `npm fund` for details
+
+found 0 vulnerabilities
+
+No new migrations to pull from main branch
+Generated Xata code to ./src/xata.ts
+
+✔ Project setup with Xata 🦋
+
+i Setup tables and columns at https://app.xata.io/workspaces/xata-workspace-hc84d7/dbs/xata-astro:eu-west-1
+
+i Use xata pull main to regenerate code and types from your Xata database
+
+```
+
+</Expand>
+
+On completion, the CLI will create `.env`, `.xatarc`, and `src/xata.ts` files within your project folder with the
+correct credentials to access your database.
+
+Your `.env` file should look something like this:
+
+```bash title=".env"
 XATA_API_KEY=YOUR_API_KEY_HERE
 XATA_BRANCH=main
 ```
 
-Since we set the project up with TypeScript support it also created files that provide typings and functions to call using Xata's TypeScript SDK. This will additionally be referenced in the `.xatarc` file as follows.
+To inform the TypeScript typechecking about these new environment variables, update `src/env.d.ts` as follows:
 
-```json {3}
+```ts title="src/env.d.ts"
+/// <reference types="astro/client" />
+
+interface ImportMetaEnv {
+  readonly XATA_API_KEY: string;
+  readonly XATA_BRANCH?: string;
+}
+
+interface ImportMeta {
+  readonly env: ImportMetaEnv;
+}
+```
+
+Since TypeScript support was selected, it also created files that provide typings and functions to call using Xata's
+[TypeScript SDK](https://xata.io/docs/typescript-client/overview). This will additionally be referenced in the `.xatarc`
+file as follows:
+
+```json
 {
-  "databaseUrl": "https://my-xata-app-database-url"
+  "databaseUrl": "https://xata-nextjs-database-url",
```

Comment:
Think this should be astro. Maybe needs a comment if you're going to change the `my` nomenclature.

Reply:
It should be `xata-{framework}`.

I adapted this from the existing guides and I'd assumed that the `my-xata-app` part was based on the name of the app being created. However, it looks like the actual format is:

```
https://{workspace-name}-workspace-{not-sure-about-this-part}.{region}.xata.sh/db/{app-name}
```

And I can't presently find any docs that cover this.

I'll revert to `my-xata-app-database-url` as that's used elsewhere.

Reply:
https://github.com/xataio/mdx-docs/pull/70/commits/eff986948d02b8ab450d99194c9405a61d66645c